### PR TITLE
[DevCenter] Data Plane v2023-04-01 .NET SDK stable, tests, and samples

### PR DESCRIPTION
**Pull request by drielenr**
# Description

We currently have a beta SDK targeting DevCenter 2023-04-01 API, and this PR is to have it released as a stable SDK - our first stable release, targeting same API. 

Both SDKs (the last one released and this one) were generated from TypeSpec. The main change introduced was add model generation and updates on tests and samples to reflect that addition. 

Arch board meeting occurred but APIView is not yet approved. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce breaking changes**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### Pipeline runs

Dry runs on the release pipeline:
https://dev.azure.com/azure-sdk/internal/_build?definitionId=5622&_a=summary

### List of tests

DevBox
 - [x] get_pool
 - [x] list_pools
 - [x] get_dev_box
 - [x] get_remote_connection
 - [x] get_action
 - [x] list_actions
 - [x] skip_action
 - [x] delay_action
 - [x] delay_all_actions
 - [x] get_schedule
 - [x] list_schedules
 - [x] list_dev_boxes
 - [x] list_all_dev_boxes
 - [x] list_all_dev_boxes_by_user
 - [x] begin_create_dev_box
 - [x] begin_delete_dev_box
 - [x] begin_start_dev_box
 - [x] begin_stop_dev_box
 - [x] begin_restart_dev_box

DevCenter
 - [x] get_project
 - [x] list_projects

Environments
 - [x] begin_delete_environment
 - [x] begin_create_or_update_environment
 - [x] list_environment_types
 - [x] list_catalogs
 - [x] list_environments
 - [x] list_all_environments
 - [x] get_environment_definition
 - [x] list_environment_definitions
 - [x]  list_environment_definitions_by_catalog
 - [x] get_catalog
 - [x] get_environment

Also updated samples.

### CODE REVIEW

File: sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxesClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Developer.DevCenter.Models;
+using Azure.Identity;
+using NUnit.Framework;
+
+namespace Azure.Developer.DevCenter.Tests
+{
+    [PlaybackOnly("As deploy/delete manipulations with real resources take time.")]
```
[the remark this answers is not in the document]
Comment (drielenr) [reply sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxesClientTests.cs:17]: oh, I see! I wasn't aware of those issues. It seems like the nightly and weekly pipeline run the same tests is this right? I don't understand the need of the weekly one if we run the same one every night.

[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxesClientTests.cs]: We run the live tests weekly to test different cloud environments. 

By default, live tests will run against Azure Public Cloud nightly, and against Azure Public Cloud and Azure Public Cloud in a canary region weekly. For tests that support sovereign cloud deployments (AzureUSGovernment and AzureChinaCloud), we will also test cloud environments weekly.

[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxesClientTests.cs]: We run the live tests weekly across different cloud environments.  By default, live tests will run against Azure Public Cloud nightly, and against Azure Public Cloud and Azure Public Cloud in a canary region weekly. For tests that support sovereign cloud deployments (AzureUSGovernment and AzureChinaCloud), we will also test cloud environments weekly.

[the remark this answers is not in the document]
Comment (drielenr) [reply sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxesClientTests.cs]: Hi @ShivangiReja. I've added the live tests, and they are all passing. See the [test pipeline](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3560365&view=results) I manually run for this branch.

**Review by sebrenna — APPROVED** [review]

**Pull request merged**